internal/tiltfile/links.go

```diff
@@ -0,0 +1,93 @@
+package tiltfile
+
+import (
+	"fmt"
+
+	"go.starlark.net/starlark"
+
+	"github.com/tilt-dev/tilt/pkg/model"
+)
+
+func convertLinks(val starlark.Value) ([]model.Link, error) {
+	// TODO: validate URL? (e.g. do we require scheme?)
+	if val == nil {
+		return nil, nil
+	}
+	switch val := val.(type) {
+	case starlark.String:
+		li, err := strToLink(val)
+		if err != nil {
+			return nil, err
+		}
+		return []model.Link{li}, nil
+	case link:
+		return []model.Link{val.Link}, nil
+	case starlark.Sequence:
+		var result []model.Link
+		it := val.Iterate()
+		defer it.Done()
+		var v starlark.Value
+		for it.Next(&v) {
+			switch v := v.(type) {
+			case starlark.String:
+				li, err := strToLink(v)
+				if err != nil {
+					return nil, err
+				}
+				result = append(result, li)
+			case link:
+				result = append(result, v.Link)
+			default:
+				return nil, fmt.Errorf("`link` arg %v includes element %v which must be a string or a link; is a %T", val, v, v)
+			}
+		}
+		return result, nil
+	default:
+		return nil, fmt.Errorf("`link` value must be an string, a link, or a sequence of those; is a %T", val)
+	}
+}
+
+func (s *tiltfileState) link(thread *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
```

Comment:
can you put this in an extension? i don't have strong opinions about which extension it is. just would like to keep the tiltfileState kitchen sink from growing

Reply:
I mean theoretically yes? but it feels weird to me to hide what is theoretically a first class Tiltfile object's constructor in an extension. I feel like we should be encouraging everyone to uses local_resource.links to name their links for self-documenting-ness, so the less friction we have there, the better. wdyt?

Reply:
i think you might be misunderstanding my comment. local_resource should eventually go in an extension too. how the code is organized shouldn't change the syntax.

Reply:
oh whoops yeah -- I thought you meant [an extension](https://github.com/tilt-dev/tilt-extensions)